clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -59,6 +60,7 @@
     private int version;
     private long lastRefreshMs;
     private long lastSuccessfulRefreshMs;
+    private ApiException nonRetriableException;
```

Comment:
From a quick glance at the code, the only case where this is populated is an authentication failure. If so, maybe we can give it a more specific name?

Reply:
I changed it to `nonRetriableAuthenticationException`. I hope that works better.